Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -124,51 +94,111 @@ extension HTTPClient {
     }
 }
 
-extension String {
-    fileprivate func omittingQuotes() -> String {
+extension StringProtocol {
+    fileprivate func omittingQuotes() -> SubSequence {
         let dquote = "\""
-        if !hasPrefix(dquote) || !hasSuffix(dquote) {
-            return self
+        guard hasPrefix(dquote), hasSuffix(dquote) else {
+            return self[startIndex..<endIndex]
         }
+        return self.dropFirst().dropLast()
+    }
 
-        let begin = index(after: startIndex)
-        let end = index(before: endIndex)
-        return String(self[begin..<end])
+    fileprivate func trimmingWhitespace() -> SubSequence {
+        guard let start = self.firstIndex(where: { !$0.isWhitespace }) else {
+            return self[self.endIndex..<self.endIndex]
+        }
+        let end = self.lastIndex(where: { !$0.isWhitespace })!
+        return self[start...end]
+    }
+
+    /// Splits this string at the first "=" sign, and trims whitespace from each side.
+    fileprivate func parseKeyValuePair() -> (key: SubSequence, value: SubSequence)? {
+        guard let keyValueSeparator = self.firstIndex(of: "=") else {
+            return nil
+        }
+        let trimmedName = self[..<keyValueSeparator].trimmingWhitespace()
+        let trimmedValue = self[self.index(after: keyValueSeparator)...].trimmingWhitespace()
+        return (trimmedName, trimmedValue)
+    }
+
+    /// If this string is a key-value pair, splits it and normalizes the key.
+    /// Otherwise, return this string, trimmed of whitespace and normalized to lowercase.
+    fileprivate func parseCookieComponent() -> (String?, String?) {
+        guard let (trimmedName, trimmedValue) = self.parseKeyValuePair() else {
+            let trimmedName = self.trimmingWhitespace()
+            return (trimmedName.isEmpty ? nil : trimmedName.lowercased(), nil)
+        }
+        guard !trimmedName.isEmpty else {
+            return (nil, nil)
+        }
+        return (trimmedName.lowercased(), trimmedValue.isEmpty ? nil : String(trimmedValue))
     }
 }
 
-extension HTTPClient.Response {
-    /// List of HTTP cookies returned by the server.
-    public var cookies: [HTTPClient.Cookie] {
-        return self.headers["set-cookie"].compactMap { HTTPClient.Cookie(header: $0, defaultDomain: self.host) }
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#endif
+import CShims
+
+private let posixLocale: locale_t = {
+    // All POSIX systems must provide a "POSIX" locale, and its date/time formats are US English.
+    // https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap07.html#tag_07_03_05
+    newlocale(LC_TIME_MASK | LC_NUMERIC_MASK, "POSIX", nil)!
+}()
+
+private func parseCookieTime(_ string: String) -> Int64? {
+    var timeComps = tm()
+    var hasExplicitTimeZone = false
+    parse: do {
+        if swiftahc_cshims_strptime_l(string, "%a, %d %b %Y %H:%M:%S %Z", &timeComps, .init(posixLocale)) != nil {
```

Comment:
Instead of putting the `init` at every call site can we just erase the type of the `posixLocale` `private let`? We never need it as a `locale_t` anyway.